client/me/privacy/dpa.jsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Card } from '@automattic/components';
+import SectionHeader from 'calypso/components/section-header';
+import { useDispatch } from 'react-redux';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import wp from 'calypso/lib/wp';
+
+const DPA = () => {
+	const [ isLoading, setLoading ] = React.useState( false );
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const requestDpa = async () => {
+		try {
+			setLoading( true );
+			await wp.req.post( '/me/request-dpa', { apiNamespace: 'wpcom/v2' } );
+			dispatch(
+				successNotice(
+					translate( 'Request successful! We are sending you our DPA via email', {
+						comment:
+							'A Data Processing Addendum (DPA) is a document to assure customers, vendors, and partners that their data handling complies with the law.',
+					} ),
+					{ id: 'request-dpa-notice' }
+				)
+			);
+		} catch ( error ) {
+			dispatch(
+				errorNotice(
+					error.error === 'too_many_requests'
+						? error.message
+						: translate( 'There was an error requesting a DPA', {
+								comment:
+									'A Data Processing Addendum (DPA) is a document to assure customers, vendors, and partners that their data handling complies with the law.',
+						  } ),
+					{ id: 'request-dpa-notice' }
+				)
+			);
+		} finally {
+			setLoading( false );
```

Comment:
This could lead to state being set on an unmounted component and would deserve an `isMounted` check to be 100% solid.

Reply:
I remember I wrote a helper a while ago for the Notifications panel (see [/apps/notifications/src/panel/helpers/use-safe.js](https://github.com/Automattic/wp-calypso/blob/trunk/apps/notifications/src/panel/helpers/use-safe.js)). Does it make sense to introduce that in Calypso as well? Just wondering what the best way moving forward is considering we're seeing more and more of these cases.

Reply:
Yes, I believe it would have a lot of use in Calypso, too. To make it more universal, it could just return the ref instead of supporting a callback:
```js
const isMounted = useIsMounted();

.then( () => {
  isMounted.current && setState( ... );
} );
```